src/base/enum_iterator.h

```diff
@@ -0,0 +1,50 @@
+#ifndef BASE_ENUM_ITERATOR_H
+#define BASE_ENUM_ITERATOR_H
+
+#include <cstddef>
+#include <type_traits>
+
+template<typename Enum, std::size_t Size = static_cast<std::size_t>(Enum::NUM)>
+class CEnumIterator
+{
+	static_assert(std::is_enum_v<Enum>, "EnumMap requires an enum type");
+
+	class CIterator
+	{
+	public:
+		CIterator(std::size_t Index) :
+			m_Index(Index) {}
+
+		CIterator &operator++()
+		{
+			++m_Index;
+			return *this;
+		}
+
+		bool operator!=(const CIterator &other) const
+		{
+			return m_Index != other.m_Index;
+		}
+
+		Enum operator*() const
+		{
+			return static_cast<Enum>(m_Index);
+		}
+
+	private:
+		std::size_t m_Index;
```

Comment:
```suggestion
		std::uint32_t m_Index;
```
I don't see why this should depend on the CPU's pointer width.

Reply:
size_t is used for array indexes, there is no enum that would exceed uint32_t

https://stackoverflow.com/questions/2550774/what-is-size-t-in-c

if you still think it should be of a fixed size, then sure